crates/runtime/src/embeddings/execution_plan.rs

```diff
@@ -314,24 +314,77 @@ pub(crate) async fn compute_additional_embedding_columns(
 /// +---------------------+      +-------------+
 ///     [`StringArray`]        [`FixedSizeListArray`]
 /// ```
+///
+/// Elements of `arr` that are `Some("")` or `None` should not be passed into
+/// [`Embed`]. This means that `arr` must be partitioned, then reconstructed
+/// after the non empty/null inputs are embedded.
+///
+/// ```text
+///                                     +- Embedding Model -+
+/// +---------------------+             |                   v
+/// | "Hello"             |      +--------------+    +------------+
+/// | None                | ---> | "Hello"      |    | [0.1, 1.2] |
+/// | ""                  |      | "Valid text" |    | [0.8, 0.9] |
+/// | "Valid text"        |      +--------------+    +------------+
+/// +---------------------+                             |  |
+///    [`StringArray`]                                  |  |
+///                   +------------+                    |  |
+///                   | [0.1, 0.2] | <------------------+  |
+///                   | None       |                       |
+///                   | None       |                       |
+///                   | [0.8, 0.9] | <---------------------+
+///                   +------------+
+///
+///                 [`FixedSizeListArray`]
+/// ```
+#[allow(clippy::cast_possible_truncation, clippy::cast_possible_wrap)]
 async fn get_vectors(
```

Comment:
This functions needs some unit tests to cover it.

Reply:
Good point, helped find an edge case too!